packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx

```diff
@@ -0,0 +1,48 @@
+import { defineComponent, toRefs } from 'vue'
+import { HexToHSV } from '../../utils/color-utils'
+import { convertToUnit, deepEqual } from '../../utils/helpers'
+import { changeColorValue } from '../../utils/composable'
+import { call, createNamespace } from '../../../utils/components'
+import { colorPickerSwatchesColorProps, ColorPickerSwatchesColorProps } from './color-picker-swatches-types'
+import './color-picker-swatches.less'
+
+export default defineComponent({
+  name: 'VarColorPickerSwatches',
+  props: colorPickerSwatchesColorProps,
+  setup(props: ColorPickerSwatchesColorProps) {
+    const { n } = createNamespace('color-picker-swatches')
+
+    const { color, maxHeight } = toRefs(props)
+
+    return () => {
+      return (
+        <div
+          class={n()}
+          style={{
+            maxHeight: convertToUnit(maxHeight!.value),
+          }}
+        >
+          <div class={n('colorful')}>
+            {props.swatches.map((swatch) => (
+              <div class={n('swatch')}>
+                {swatch.map((colorItem) => {
+                  const hsv = HexToHSV(colorItem)
+                  const iconColor = changeColorValue(hsv!, 0.5)
+                  return (
+                    <div class={n('color')} onClick={() => call(props['onUpdate:color'], hsv)}>
+                      <div style={{ background: colorItem }}>
+                        {color!.value && deepEqual(color!.value, hsv) ? (
+                          <var-icon size="15" name="checkbox-marked-circle" color={iconColor.color} />
+                        ) : undefined}
```

Comment:
{color!.value && deepEqual(color!.value, hsv) &&

Reply:
copy that

Reply:
del this branch restart new pr